mythtv/libs/libmythtv/mythplayer.cpp

```diff
@@ -661,16 +661,13 @@ void MythPlayer::DiscardVideoFrames(bool KeyFrame, bool Flushed)
 
 bool MythPlayer::HasReachedEof(void) const
 {
-    EofState eof = GetEof();
-    if (eof != kEofStateNone && !m_allPaused)
-        return true;
     if (GetEditMode())
         return false;
     if (m_liveTV)
         return false;
     if (!m_deleteMap.IsEmpty() && m_framesPlayed >= m_deleteMap.GetLastFrame())
         return true;
-    return false;
+    return (GetEof() != kEofStateNone && !m_allPaused);
```

Comment:
This can return different results than before. Try the combination of eof is kEofStateDelayed, m_allPaused is false, and GetEditMode() is true. Before the change this would return true, after the change it returns false.

Reply:
Good point, I'll have to look into it more deeply, but my intuition is that you wouldn't want to exit because of EOF while editing the recording.

It may also make sense to add:
```
if (GetEof() == kEofStateImmediate)
    return true;
```

I'll investigate further.